ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -4195,21 +4195,18 @@ public static long unsetBit(long bitmap, int bitIdx) {
    * DISTINCT, if present, will be handled when generating the SELECT.
    */
   List<ASTNode> getGroupByForClause(QBParseInfo parseInfo, String dest) throws SemanticException {
-    // When *not* invoked by CalcitePlanner, return the DISTINCT as a GBY
-    // CBO will handle the DISTINCT in CalcitePlannerAction.genSelectLogicalPlan
     ASTNode selectExpr = parseInfo.getSelForClause(dest);
     Collection<ASTNode> aggregateFunction = parseInfo.getDestToAggregationExprs().get(dest).values();
-    if (isSelectDistinct(selectExpr) && !isGroupBy(selectExpr) && !isAggregateInSelect(selectExpr, aggregateFunction)) {
+    if (isSelectDistinct(selectExpr) && !hasGroupBySibling(selectExpr) &&
+        !isAggregateInSelect(selectExpr, aggregateFunction)) {
       List<ASTNode> result = new ArrayList<ASTNode>(selectExpr == null ? 0 : selectExpr.getChildCount());
```

Comment:
remove this `== null` check ; it's not needed

Reply:
removed